source/daplink/interface/swd_host.c

```diff
@@ -843,7 +843,10 @@ __attribute__((weak)) void swd_set_target_reset(uint8_t asserted)
 uint8_t swd_set_target_state_hw(TARGET_RESET_STATE state)
 {
     uint32_t val;
-    swd_init();
+
+    if (state != RUN) {
```

Comment:
It may be good to leave a note explaining that calling swd_init repeatedly causes operations to fail.

Reply:
I have added a comment.